rclrs/src/wait/guard_condition.rs

```diff
@@ -0,0 +1,133 @@
+use crate::rcl_bindings::*;
+use crate::{Context, RclrsError, ToResult};
+
+use std::sync::{atomic::AtomicBool, Arc, Mutex, MutexGuard};
+
+/// A struct for encapsulating a guard condition - a waitable trigger
+pub struct GuardCondition {
+    /// The rcl_guard_condition_t that this struct encapsulates.
+    rcl_guard_condition: Arc<Mutex<rcl_guard_condition_t>>,
+    /// An optional callback to call when this guard condition is triggered.
+    callback: Option<Box<dyn Fn(usize)>>,
+    /// A flag to indicate if this guard condition has already been assigned to a wait set.
+    pub(crate) in_use_by_wait_set: Arc<AtomicBool>,
+    /// A count for the number of times this guard condition was triggered, but no callback was assigned.
+    unread_count: usize,
+}
+
+impl Drop for GuardCondition {
```

Comment:
Users will probably want to use the `GuardConditions` from a separate thread than the one the wait set is running in. So we should check if we can implement `Sync` and `Send`.

Reply:
I found that in my demo, I also needed a `Send` bound on the callback.